internal/controller/apiextensions/composite/composition_functions.go

```diff
@@ -475,32 +515,12 @@ func (c *FunctionComposer) Compose(ctx context.Context, xr *composite.Unstructur
 	// NOTE(phisco): Here we are fine using a hardcoded field owner as there is
 	// no risk of conflict between different XRs.
 	if err := c.client.Status().Patch(ctx, xr, client.Apply, client.ForceOwnership, client.FieldOwner(FieldOwnerXR)); err != nil {
+		// Note(phisco): here we are fine with this error being terminal, as
+		// there is no other resource to apply that might eventually resolve
+		// this issue.
```

Comment:
It's not extremely clear from the diff, so let me make it explicit, I moved applying the XR status at the bottom to mirror the pt composer's behaviour, also because it could again be a source of issues with eventually consistent compositions to prevent composed resources to be applied due to an invalid composite status, which might become valid once composed resources are applied.

Reply:
> t could again be a source of issues with eventually consistent compositions to prevent composed resources to be applied due to an invalid composite status

@phisco What's an example of an invalid XR status that might eventually become valid? I'm wondering whether this is a real concern.

Here's two I can think of:

1. A status field is required (per the XRD OpenAPI schema), but is not yet being set.
2. A status field is a particular type (per the XRD OpenAPI schema), but you're trying to set the wrong type.

I think 2 would block forever regardless of when we update the XR status. If you're setting an invalid type, it's presumably not going to become valid when one of the composed resources was updated. Or if it does, that seems like your Composition probably still has a bug, right?

I could imagine 1 resolving itself when a composed resource was created/ready (e.g. because you're deriving the status from a composed resource). However, I think it's nonsensical/an anti-pattern to have a required status field in the first place. If and when we have a guide to writing XRDs (https://github.com/crossplane/docs/issues/564) I'd include this.

Reply:
Could be a required field, but also a validation rule, or an integer with a minimum value. I agree it doesn't make much sense to set those in the status, probably, but I don't see any reason not to do it now that I noticed it 😅

Reply:
Invalid composed resources are not blocking now (with this patch). Did you mean other types of errors?

Reply:
> Did you mean other types of errors?

Yeah, either transient errors (e.g. network connectivity etc) or something that requires human intervention to fix (e.g. RBAC issues)